entity.go

```diff
@@ -173,7 +173,9 @@ type IndexDefinition struct {
 // Clone returns a deep copy of IndexDefinition
 func (id *IndexDefinition) Clone() *IndexDefinition {
 	var columns []string
```

Comment:
We could fix this by instead saying `columns := make([]string, len(id.Columns)` right?

Reply:
tried to use make to construct the string array before when I was debugging the string(nil) and string{} issue. The key point is that. when the upsert get into Schema service. it's stored as nil but not []string{}, so, after the first upsert, when u check upsertable again, index incompatible happened as well. And I believe it's also the way why all the previous Clone are using  nil and append.
it's super tricky, and hard to debug. didn't find any comment and explanation. jump into the trap is the only way to know..